cpp/include/cudf/strings/strings_column_view.hpp

```diff
@@ -0,0 +1,476 @@
+    *---------------------------------------------------------------------------**/
+    const bitmask_type* null_mask() const;
+
+    /**---------------------------------------------------------------------------*
+    * @brief Returns the number of nulls in this column
+    *---------------------------------------------------------------------------**/
+    size_type null_count() const;
+
+private:
+    const column_view _parent;
+};
+
+namespace strings
+{
+
+/**---------------------------------------------------------------------------*
+ * @brief Prints the strings to stdout.
+ *
+ * @param strings Strings instance for this operation.
+ * @param start Index of first string to print.
+ * @param end Index of last string to print. Specify -1 for all strings.
+ * @param max_width Maximum number of characters to print per string.
+ *        Specify -1 to print all characters.
+ * @param delimiter The chars to print between each string.
+ *        Default is new-line character.
+ *---------------------------------------------------------------------------**/
+void print( strings_column_view strings,
+            size_type start=0, size_type end=-1,
+            size_type max_width=-1, const char* delimiter = "\n" );
+
+
+/**---------------------------------------------------------------------------*
+ * @brief Create output pair per Arrow format of strings.
+ * The return pair is a the array of chars and an array of offsets.
+ *
+ * @param strings Strings instance for this operation.
+ * @param stream CUDA stream to use kernels in this method.
+ * @param mr Resource for allocating device memory.
+ * @return Contiguous array of chars and an array of offsets.
+ *---------------------------------------------------------------------------**/
+std::pair<rmm::device_vector<char>, rmm::device_vector<size_type>>
+    create_offsets( strings_column_view strings,
+                    cudaStream_t stream=0,
+                    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource() );
+
+// array.cu
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column created from a subset of
+ * of this instance's strings column.
+ *
+ * @code
+ * s1 = ["a", "b", "c", "d", "e", "f"]
+ * s2 = sublist( s1, 2 )
+ * s2 is ["c", "d", "e", "f"]
+ * @endcode
+ *
+ * @param strings Strings instance for this operation.
+ * @param start Index of first string to use.
+ * @param end Index of last string to use.
+ *        Default -1 indicates the last element.
+ * @param step Increment value between indexes.
+ *        Default step is 1.
+ * @param stream CUDA stream to use kernels in this method.
+ * @param mr Resource for allocating device memory.
+ * @return New strings column of size (end-start)/step.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> sublist( strings_column_view strings,
```

Comment:
The `cudf::slice` method does not appear to have the same behavior.
It is likely I'm missing something. 
The `gather` and `sublist` methods here are to handle the many, many number of ways `__get_item__` can be called in python.

Reply:
As discussed in the PR comments, we should aim to have a single interface to cuDF columns and algorithms, not separate APIs for string columns and all other columns.  If cudf::slice can't be used to slice string columns, we'll need to fix it so it can.